addons/binding/org.openhab.binding.omnikinverter/src/main/java/org/openhab/binding/omnikinverter/OmnikInverter.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.omnikinverter;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Hans van den Bogert - Initial Contribution
+ *
+ */
+public class OmnikInverter {
+    private final Logger logger = LoggerFactory.getLogger(OmnikInverter.class);
+
+    int serialNumber;
+    String host;
+    int port;
+
+    public OmnikInverter(String host, int port, int serialNumber) {
+        this.host = host;
+        this.port = port;
+        this.serialNumber = serialNumber;
+    }
+
+    public OmnikInverterMessage pullCurrentStats() throws UnknownHostException, IOException {
+        byte[] magicPacket = generateMagicPacket();
+        byte[] returnMessage = new byte[1024];
+        Socket socket = new Socket(host, port);
+        socket.setSoTimeout(5000);
+        socket.getOutputStream().write(magicPacket);
+        socket.getInputStream().read(returnMessage);
+        socket.close();
+        return new OmnikInverterMessage(returnMessage);
+    }
+
+    private byte[] generateMagicPacket() {
+        Byte[] byteArr = new Byte[5];
```

Comment:
Seems this variable is unused.

Reply:
hmm yeah, was early code which mistakenly got committed. A pity eclipse didn't catch that.